src/action/base/add_user_to_group.rs

```diff
@@ -0,0 +1,284 @@
+                return Err(ActionError::UserGidMismatch(
+                    name.clone(),
+                    user.gid.as_raw(),
+                    gid,
+                ));
+            }
+
+            // See if group membership needs to be done
+            match target_lexicon::OperatingSystem::host() {
+                OperatingSystem::MacOSX {
+                    major: _,
+                    minor: _,
+                    patch: _,
+                }
+                | OperatingSystem::Darwin => {
+                    let mut command = Command::new("/usr/sbin/dseditgroup");
+                    command.process_group(0);
+                    command.args(["-o", "checkmember", "-m"]);
+                    command.arg(&this.name);
+                    command.arg(&this.groupname);
+                    command.stdout(Stdio::piped());
+                    command.stderr(Stdio::piped());
+                    let command_str = format!("{:?}", command.as_std());
+                    tracing::trace!("Executing `{command_str}`");
+                    let output = command.output().await.map_err(ActionError::Command)?;
+                    match output.status.code() {
+                        Some(0) => {
+                            // yes {user} is a member of {groupname}
+                            // Since the user exists, and is already a member of the group, we have truly nothing to do here
+                            tracing::debug!(
+                                "Adding user `{}` to group `{}` already complete",
+                                this.name,
+                                this.groupname
+                            );
+                            return Ok(StatefulAction::completed(this));
+                        },
+                        Some(64) => {
+                            // Group not found
+                            tracing::trace!(
+                                "Will add user `{}` to newly created group `{}`",
+                                this.name,
+                                this.groupname
+                            );
+                            // The group will be created by the installer
+                            ()
+                        },
+                        _ => {
+                            // Some other issue
+                            return Err(ActionError::Command(std::io::Error::new(
+                                std::io::ErrorKind::Other,
+                                format!(
+                                    "Command `{command_str}` failed{}, stderr:\n{}\n",
+                                    if let Some(code) = output.status.code() {
+                                        format!(" status {code}")
+                                    } else {
+                                        "".to_string()
+                                    },
+                                    String::from_utf8(output.stderr)
```

Comment:
I wonder if it would be a good idea to try to recover the non-utf8 output by using [`String::from_utf8_lossy`](https://doc.rust-lang.org/std/string/struct.String.html#method.from_utf8_lossy)?

Reply:
What would we do? Print it raw?

Reply:
Yeah, with replacement characters and all.

Not really necessary, but I figure it's better to see whatever could be converted to utf8 and not just "UTF-8 conversion error" or similar.